benchmarking/deepcell-e2e/deepcell-e2e-benchmark-install.ipynb

```diff
@@ -0,0 +1,436 @@
+    "    ax.set_title('Predictions')\n",
+    "    plt.show()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "c598730f-4762-4c1a-ba2c-02698632c039",
+   "metadata": {},
+   "source": [
+    "# Benchmark data"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "id": "4b4cca55-1a31-4803-ad62-ac3c3c45d75c",
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "headers = [\n",
+    "    'Input file id', 'File size (MB)', 'Benchmark datetime (UTC)',\n",
+    "    'Machine type', 'GPU type', '# GPUs',\n",
+    "    'Success?', 'Total time (s)', 'Peak memory (GB)',\n",
+    "    'Load time (s)', 'Total prediction time (s)', 'Prediction overhead (s)',\n",
+    "    'Predict preprocess time (s)', 'Predict inference time (s)', 'Predict postprocess time (s)',\n",
+    "    'deepcell-tf version'\n",
+    "]\n",
+    "\n",
+    "parsed_url = urllib.parse.urlparse(input_channels_path)\n",
+    "filename = parsed_url.path.split(\"/\")[-2]\n",
+    "file_size = round(input_channels.nbytes / 1000 / 1000, 2)\n",
+    "\n",
+    "if notebook_instance_id:\n",
+    "    # For running on vertex AI:\n",
+    "    # Shell command:\n",
+    "    # $ gcloud notebooks runtimes describe --format=\"value(virtualMachine.virtualMachineConfig.machineType)\" --location=<region> <notebook_id>\n",
+    "    machine_type = 'todo-machine'\n",
+    "else:\n",
+    "    # assume a generic python environment\n",
+    "    # See also:\n",
+    "    # https://docs.python.org/3.10/library/os.html#os.cpu_count\n",
+    "    try:\n",
+    "        num_cpus = len(os.sched_getaffinity(0))\n",
+    "    except AttributeError:\n",
+    "        num_cpus = os.cpu_count()\n",
+    "    total_mem = psutil.virtual_memory().total\n",
+    "    machine_type = 'local {} CPUs {} GB RAM'.format(num_cpus, round(total_mem/1000000000, 1))\n",
+    "\n",
+    "peak_mem_b = resource.getrusage(resource.RUSAGE_SELF).ru_maxrss\n",
+    "prediction_overhead_s = prediction_time_s - preprocess_time_s - inference_time_s - postprocess_time_s\n",
+    "\n",
+    "# Write benchmarking data as CSV:\n",
+    "\n",
+    "output = io.StringIO()\n",
+    "writer = csv.writer(output, quoting=csv.QUOTE_NONNUMERIC)\n",
+    "writer.writerow(headers)\n",
+    "\n",
+    "deepcell_version = deepcell.__version__\n",
+    "\n",
+    "writer.writerow([\n",
+    "    filename, file_size, datetime.now(timezone.utc),\n",
+    "    machine_type, 'todo-gpu-type', 'todo-gpu-count',\n",
+    "    prediction_success, round(total_time_s, 2), round(peak_mem_b / 1000000000, 1),\n",
+    "    round(input_load_time_s, 2), round(prediction_time_s, 2), round(prediction_overhead_s, 2),\n",
+    "    round(preprocess_time_s, 2), round(inference_time_s, 2), round(postprocess_time_s, 2),\n",
+    "    deepcell_version,\n",
+    "])\n",
+    "\n",
+    "print(output.getvalue())"
+   ]
+  }
+ ],
+ "metadata": {
+  "kernelspec": {
+   "display_name": "TensorFlow 2.12 (Local)",
```

Comment:
qq @lynnlangit  – do you know if this metadata makes vertex AI actually select the kernel? Or is it recording the kernel you had when you ran it?

Reply:
if you don't pip install deepcell in a cell in the nb, the import statement for deepcell in the e-2-e nb fails.
because there are 3 versions of the TF runtime available in VertexAI, we should note in the testing notebook which version we are using for testing.  You must manually select the runtime.

Reply:
Shame– I was hoping setting the metadata "Just Did It". Speaking of, I think DeepCell only works with 2.8 😞  

See also: https://github.com/vanvalenlab/deepcell-tf/blob/master/requirements.txt

So I'm actually not 100% sure what the TF kernel does for us? If we're ending up reinstalling TF2.8 via the dependencies? 🤔

Reply:
I think compiling with latest is not enough alone: iirc the model will need to be retrained as well.

My guess is productionizing would warrant the container, if nothing else to avoid reinstalling the half-gig TF 2.8 dependency each time. For our purposes we may be able to get away with this.

So I'm thinking: let's put on some blinders and see if we can proceed ... And if not, we have those 2 options: (1) version upgrade & associated work, and/or, (2) use a container.

Fwiw the container approach does let us specify parameters & instance types programmatically. I think, at least 😂